sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageExtensions.cs

```diff
@@ -26,9 +27,21 @@ private static IEnumerable<Data> AsAmqpData(this IEnumerable<ReadOnlyMemory<byte
         {
             foreach (ReadOnlyMemory<byte> data in binaryData)
             {
+                ArraySegment<byte> segment;
+                if (!data.IsEmpty)
+                {
+                    if (!MemoryMarshal.TryGetArray(data, out segment))
+                    {
+                        segment = new ArraySegment<byte>(data.ToArray());
```

Comment:
this is also for safety reasons so that we could fall back on the more allocation heavy path if needed